src/commands/new.rs

```diff
@@ -28,21 +44,39 @@ impl NewCommand {
         self.run_template(repo, id).await
     }
 
+    async fn list_templates(&self) -> anyhow::Result<()> {
+        let entries = hub_api::index().await.unwrap();
+        let matches = entries.iter()
+            .filter(|e| self.is_terms_match(e))
+            .sorted_by_key(|e| e.title())
+            .collect_vec();
+
+        if matches.is_empty() {
+            println!("No templates match your search terms");
+            return Ok(());
+        }
+
+        for entry in matches {
+            println!("Template: {}\nDescription: {}\n", entry.title(), entry.summary());
+        }
+
+        Ok(())
+    }
+
     async fn run_template(&self, repo: String, id: String) -> anyhow::Result<()> {
         use spin_templates::*;
 
-        let tempdir = tempfile::tempdir().unwrap();
-        let template_manager = spin_templates::TemplateManager::in_dir(tempdir.path());
+        let manager = TemplateManager::try_default()?;
```

Comment:
If you do this, the entire repo of templates will be left installed in the user's system (and raises issues of what to do if the user has the template already).  Using a TemplateManager in a temporary directory means we can pull the templates for the duration of the command without affecting the user's installed templates.